src/commands/upload_dart_symbol_map.rs

```diff
@@ -0,0 +1,167 @@
+}
+
+impl AsRef<[u8]> for DartSymbolMapObject {
+    fn as_ref(&self) -> &[u8] {
+        &self.bytes
+    }
+}
+
+impl Display for DartSymbolMapObject {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "dartsymbolmap {}", self.name)
+    }
+}
+
+impl Assemblable for DartSymbolMapObject {
+    fn name(&self) -> Cow<'_, str> {
+        Cow::from(self.name.as_str())
+    }
+
+    fn debug_id(&self) -> Option<DebugId> {
+        Some(self.debug_id)
+    }
+}
+
+pub fn make_command(command: Command) -> Command {
+    command
+        .about("Upload a Dart/Flutter symbol map (dartsymbolmap) for deobfuscating Dart exception types.")
+        .after_help(
+            "Examples:\n  sentry-cli dart-symbol-map upload --org my-org --project my-proj path/to/dartsymbolmap.json path/to/debug/file\n\n  The mapping must be a JSON array of strings with an even number of entries (pairs).\n  The debug file must contain exactly one Debug ID.",
+        )
+        .org_arg()
+        .project_arg(false)
+        .arg(
+            Arg::new("mapping")
+                .value_name("MAPPING")
+                .help("Path to the dartsymbolmap JSON file (e.g. dartsymbolmap.json). Must be a JSON array of strings with an even number of entries (pairs).")
+                .required(true),
+        )
+        .arg(
+            Arg::new("debug_file")
+                .value_name("DEBUG_FILE")
+                .help("Path to the corresponding debug file to extract the Debug ID from. The file must contain exactly one Debug ID.")
+                .required(true),
+        )
+}
+
+pub fn execute(matches: &ArgMatches) -> Result<()> {
+    // Parse required positional arguments
+    let mapping_path = matches
+        .get_one::<String>("mapping")
+        .expect("required argument 'mapping' not provided by clap");
+    let debug_file_path = matches
+        .get_one::<String>("debug_file")
+        .expect("required argument 'debug_file' not provided by clap");
+
+    // Extract Debug ID(s) from the provided debug file
+    let dif = DifFile::open_path(debug_file_path, None)?;
+    let mut ids: Vec<_> = dif.ids().into_iter().filter(|id| !id.is_nil()).collect();
+
+    // Ensure a single, unambiguous Debug ID
+    ids.sort();
+    ids.dedup();
+    match ids.len() {
+        0 => bail!(
+            "No debug identifier found in the provided debug file ({}). Ensure the file contains an embedded Debug ID.",
+            debug_file_path
+        ),
+        1 => {
+            let debug_id = ids.remove(0);
+
+            // Validate the dartsymbolmap JSON: must be a JSON array of strings with even length
+            let mapping_file_bytes = std::fs::read(mapping_path)
+                .with_context(|| format!("Failed to read mapping file at {mapping_path}"))?;
+            let mapping_entries: Vec<String> = serde_json::from_slice(&mapping_file_bytes)
+                .context("Invalid dartsymbolmap: expected a JSON array of strings")?;
```

Comment:
This line, as currently written, will also copy over the entire contents of each entry into memory as `String`s. Instead, we should be able to use `&str` to ensure no copies of entries are made, and we instead directly reference the memory-mapped bytes

```suggestion
            let mapping_entries: Vec<&str> = serde_json::from_slice(&mapping_file_bytes)
                .context("Invalid dartsymbolmap: expected a JSON array of strings")?;
```